common/djangoapps/enrollment/views.py

```diff
@@ -135,7 +135,9 @@ def get(self, request, course_id=None, username=None):
         """
         username = username or request.user.username
 
-        if request.user.username != username and not self.has_api_key_permissions(request):
+        # TODO Implement proper permissions
```

Comment:
FYI there's [this](https://github.com/edx/edx-platform/blob/master/openedx/core/lib/api/permissions.py)